File: ruby/prime-factors/prime_factors.rb
class PrimeFactors
  def self.for(num)
    curr = 2
    [].tap do |factors|
      while num > 1
        if num % curr == 0
          num = num / curr
          factors << curr
        else
          curr +=1
        end
      end
    end
  end
end

File: javascript/prime-factors/prime-factors.js
var primeFactors = function(num) {
  var curr = 2;
  var factors = [];
  while (num > 1) {
    if (num % curr == 0) {
      num = (num / curr);
      factors.push(curr);
    } else {
      curr += 1;
    }
  }
  return factors;
};

module.exports = { for: primeFactors };
